server/api/utils/clients/iguazio.py

```diff
@@ -783,6 +926,25 @@ def _handle_error_response(
         self._logger.warning("Request to iguazio failed", **log_kwargs)
         mlrun.errors.raise_for_status(response, error_message)
 
+    def _is_job_done(self, session: str, job_id: str) -> bool:
+        """
+        Check if the iguazio job is done
+
+        :param session: iguazio session
+        :param job_id: iguazio job id
+        :return: True if the job is done, False otherwise
+        """
+        try:
+            response = self._get_job_from_iguazio(session, job_id)
+            return (
+                response["data"]["attributes"]["state"] in JobStates.terminal_states()
+            )
+        except requests.HTTPError as exc:
+            if exc.response.status_code == http.HTTPStatus.NOT_FOUND.value:
+                # job not found - consider it as done
+                return True
```

Comment:
I don't think you should decide the 404 is done here. You should raise mlrun not found error and decide what you want to do outside